tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +137,139 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
 	await expect(page).toHaveURL(`/onboarding`)
 })
 
+test('onboarding with GitHub OAuth', async ({ page }) => {
+	// grab mock github users from tests/fixtures/github
+	const ghUsers = await getGitHubUsers()
+	const ghUser = ghUsers.find(u => u.code === MOCK_CODE_GITHUB)
+
+	// a github user can have numerous emails, but only one is primary,
+	// so we use it to query our users relation: if a user with this email
+	// is there, we simply create a link between the 'user' and  'connection'
+	// entities and make a new session, since the user is already on board;
+	// this behavior will be tested later on, but for now we need to prepare the ground:
+	const user = await prisma.user.findUnique({
+		where: { email: ghUser?.primaryEmail.toLowerCase() },
+	})
+	if (user) await prisma.user.delete({ where: { id: user.id } })
+
+	// we remove the connection if any to keep the test clean
+	const conn = await prisma.connection.findFirst({
+		where: { providerName: 'github', userId: user?.id },
+	})
+	if (conn) await prisma.connection.delete({ where: { id: conn.id } })
```

Comment:
Rather than finding an existing user and deleting them, I would prefer to restructure things so we create a brand new github user for this test.

Reply:
That was my original intention: each test should be isolated and have its own prep and clean up steps that will not impact the other tests, that's true. But I am now struggling a bit with the [code](https://github.com/epicweb-dev/epic-stack/blob/main/app/utils/providers/github.server.ts#L87) being a predefined value in  the `handleMockAction` method, and we're going to look up a user from our mocked github users database using this value. I am looking into it

Reply:
All right, playwright gives us API to intercept requests and patch them, so used this approach. This gives us proper isolation. 

Just for the sake of discussion. The other option would be to generate (rather than have it hard coded) a `code` in the `mockHandleAction` method, insert a user with this code into the mocked github users database, and then redirect with this code in query string.